opendevin/llm/llm.py

```diff
@@ -1,20 +1,21 @@
-from litellm import completion
+from litellm import completion as litellm_completion
+from functools import partial
 import os
 
 DEFAULT_MODEL = os.getenv("LLM_MODEL", "gpt-4-0125-preview")
+DEFAULT_API_KEY = os.getenv("LLM_API_KEY")
 
 class LLM:
-    def __init__(self, model=DEFAULT_MODEL):
-        self.model = model
-        if self.model == "" or self.model is None:
-            self.model = DEFAULT_MODEL
+    def __init__(self, model=DEFAULT_MODEL, api_key=DEFAULT_API_KEY):
+        self.model = model if model else DEFAULT_MODEL
+        self.api_key = api_key if api_key else DEFAULT_API_KEY
 
-    def prompt_with_messages(self, messages, args: dict={}):
-        if self.model == 'fake':
-            return "This is a fake response"
-        resp = completion(model=self.model, messages=messages, **args)
-        return resp['choices'][0]['message']['content']
+        self._completion = partial(litellm_completion, model=self.model, api_key=self.api_key)
```

Comment:
@rbren here's what i proposed potential solution we can use for an thin LLM wrapper - we just supply the model name, API keys etc to the llm instance; And inside `Agent` people can do whatever they want with the `completion` function, since it has the exact same signature of litellm (but they don't need to handle the temperature etc) since it is just a very thin wrapper. Feel free to revert this commit if you have concerns :)

Reply:
Oooh `partial` is neat! I like this solution.